crates/telio-proxy/src/proxy.rs

```diff
@@ -187,10 +208,61 @@ impl StateEgress {
 
         self.wg_addr = wg_port.map(|p| (Ipv4Addr::LOCALHOST, p).into());
 
-        self.sockets = sockets;
+        let old_sockets = self.sockets.clone();
+        self.sockets = sockets
```

Comment:
Shouldn't you clean up the `self.mutes` of the sockets that are no longer present in `self.sockets` as well?